### PR TITLE
Add cross compile for A311D

### PR DESCRIPTION
**Pull request by ZongwuYang**
* Support auto-downloading toolchain and SDK from http
  * Support one-click cross compilation

Usage:

```bash
  mkdir build && cd build
  cmake .. -DHARDWARE=A311D
  make -j4
```

Signed-off-by: Zongwu.Yang <Zongwu.Yang@verisilicon.com>

### CODE REVIEW

File: cmake/OvxDrvConfig.cmake [hunk 1]
```diff
@@ -3,27 +3,45 @@ message("Processing OvxDrvConfig.cmake ...")
...
-    ${PROJECT_SOURCE_DIR}/prebuilt-sdk/x86_64_linux/include/CL)
 
-set(LIBS CLC GAL OpenVX OpenVXU VSC ArchModelSw NNArchPerf)
-if(CMAKE_CROSSCOMPILING)
-  # todo (sven) add any library if only avaiable in cross build
+if("${HARDWARE}" STREQUAL "A311D")
+    list(APPEND OVXDRV_INCLUDE_DIRS
+        ${PROJECT_SOURCE_DIR}/build/aarch64_A311D_D312513_A294074_R311680_T312233_O312045/include
+        ${PROJECT_SOURCE_DIR}/build/aarch64_A311D_D312513_A294074_R311680_T312233_O312045/include/CL)
+elseif("${HARDWARE}" STREQUAL "NXP865")
```
Comment (thezha) [inline cmake/OvxDrvConfig.cmake:11]: Please remove this.

File: cmake/OvxDrvConfig.cmake [hunk 2]
```diff
@@ -3,27 +3,45 @@ message("Processing OvxDrvConfig.cmake ...")
...
-  message("--->" ${PKG_NAME}_LIBRARIES ${${PKG_NAME}_FOUND_LIB_${i}})
+  if("${HARDWARE}" STREQUAL "A311D")
+    find_library(${PKG_NAME}_FOUND_LIB_${i}
+        NAMES ${lib}
+        PATHS ${PROJECT_SOURCE_DIR}/build/aarch64_A311D_D312513_A294074_R311680_T312233_O312045/lib
+    )
+    message("found lib ${lib} at ${${PKG_NAME}_FOUND_LIB_${i}}")
+    list(APPEND ${PKG_NAME}_LIBRARIES ${${PKG_NAME}_FOUND_LIB_${i}})
+    message("--->" ${PKG_NAME}_LIBRARIES ${${PKG_NAME}_FOUND_LIB_${i}})
+  elseif("${HARDWARE}" STREQUAL "NXP865")
```
Comment (thezha) [inline cmake/OvxDrvConfig.cmake:33]: Please remove this.

File: cmake/OvxDrvConfig.cmake [hunk 3]
```diff
@@ -3,27 +3,45 @@ message("Processing OvxDrvConfig.cmake ...")
...
 
-set(LIBS CLC GAL OpenVX OpenVXU VSC ArchModelSw NNArchPerf)
-if(CMAKE_CROSSCOMPILING)
-  # todo (sven) add any library if only avaiable in cross build
+if("${HARDWARE}" STREQUAL "A311D")
+    list(APPEND OVXDRV_INCLUDE_DIRS
+        ${PROJECT_SOURCE_DIR}/build/aarch64_A311D_D312513_A294074_R311680_T312233_O312045/include
+        ${PROJECT_SOURCE_DIR}/build/aarch64_A311D_D312513_A294074_R311680_T312233_O312045/include/CL)
+elseif("${HARDWARE}" STREQUAL "NXP865")
+    message("Need set nxp865 sdk include.")
```
Comment (thezha) [inline cmake/OvxDrvConfig.cmake:12]: Remove all m865 related references for now.

File: cmake/A311D.cmake [hunk 4]
```diff
@@ -0,0 +1,22 @@
+message("Downloading A311D toolchain ...")
+file(DOWNLOAD  "https://cnbj1.fds.api.xiaomi.com/mace/third-party/gcc-linaro/gcc-linaro-7.3.1-2018.05-x86_64_aarch64-linux-gnu.tar.xz"
+    ${PROJECT_SOURCE_DIR}/build/gcc-linaro-7.3.1-2018.05-x86_64_aarch64-linux-gnu.tar.xz
+    EXPECTED_MD5 "6ec789d642584a01e240ab3366599dbb"
+    SHOW_PROGRESS)
+execute_process(COMMAND
+    tar xvf ${PROJECT_SOURCE_DIR}/build/gcc-linaro-7.3.1-2018.05-x86_64_aarch64-linux-gnu.tar.xz)
```
Comment (hawk081) [inline cmake/A311D.cmake:7]: please help to remove the `v` option from `tar` command.

File: cmake/A311D.cmake [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+set(CMAKE_AS ${TOOLCHAIN_DIR}/bin/aarch64-linux-gnu-gcc-as)
+set(CMAKE_LD ${TOOLCHAIN_DIR}/bin/aarch64-linux-gnu-gcc-ld)
+
+message("Downloading A311D SDK ...")
+file(DOWNLOAD "https://github.com/VeriSilicon/TIM-VX/releases/download/v1.1.28/aarch64_A311D_D312513_A294074_R311680_T312233_O312045.tgz"
+    ${PROJECT_SOURCE_DIR}/build/aarch64_A311D_D312513_A294074_R311680_T312233_O312045.tgz
+    EXPECTED_MD5 "26f4766c25a4776caf41b93e9b1c6af9"
+    SHOW_PROGRESS)
+execute_process(COMMAND 
+    tar xvf ${PROJECT_SOURCE_DIR}/build/aarch64_A311D_D312513_A294074_R311680_T312233_O312045.tgz)
```
Comment (hawk081) [inline cmake/A311D.cmake:22]: please help to remove the `v` option from `tar` command.

**Review by hawk081 — CHANGES_REQUESTED** [review]

**Pull request merged**